extendr-api/src/scalar/macros.rs

```diff
@@ -544,61 +544,6 @@ macro_rules! gen_from_primitive {
     };
 }
 
-/// Generates an implementation of type conversion Traits from a scalar type
-///
-/// This macro requires the following arguments:
-///
-/// * `$type`      - The Type the unary operator Trait is implemented for
-/// * `$type_prim` - The primitive Rust scalar type that corresponds to `$type`
-///
-/// Example Usage:
-///
-/// ```ignore
-/// gen_from_scalar!(Rint, i32);
-/// ```
-///
-/// The 'example usage' implements the following trait definitions:
-///
-/// - `From<Rint> for Option<i32>`
-/// - `From<Rint> for Robj`
```

Comment:
It's good that the new impl_synonym_type macro can replace this with and improve upon it by adding method for each default. However, it is very nice that this macro has documentation. Can you replicate the documentation for the new macro ?